src/features/lightspeed/vue/views/ansibleCreatorUtils.ts

```diff
@@ -0,0 +1,422 @@
+    } else {
+      ansibleCreatorAddCommand += " --no-overwrite";
+    }
+
+    const verbosityMap: Record<string, string> = {
+      off: "",
+      low: " -v",
+      medium: " -vv",
+      high: " -vvv",
+    };
+
+    const normalizedVerbosity = verbosity.toLowerCase();
+    const verbosityFlag = verbosityMap[normalizedVerbosity] || "";
+    ansibleCreatorAddCommand += verbosityFlag;
+
+    console.debug("[ansible-creator] command: ", ansibleCreatorAddCommand);
+
+    const extSettings = new SettingsManager();
+    await extSettings.initialize();
+
+    const { command, env } = withInterpreter(
+      extSettings.settings,
+      ansibleCreatorAddCommand,
+      "",
+    );
+
+    let commandOutput = "";
+    let commandResult: string;
+
+    const creatorVersion = await getCreatorVersion();
+    const minRequiredCreatorVersion: Record<string, string> = {
+      role: "25.4.0",
+    };
+    const requiredCreatorVersion = minRequiredCreatorVersion["role"];
+
+    commandOutput += `----------------------------------------- ansible-creator logs ------------------------------------------\n`;
+
+    if (semver.gte(creatorVersion, requiredCreatorVersion)) {
+      const ansibleCreatorExecutionResult = await runCommand(command, env);
+      commandOutput += ansibleCreatorExecutionResult.output;
+      commandResult = ansibleCreatorExecutionResult.status;
+    } else {
+      commandOutput += `Minimum ansible-creator version needed to add the role resource is ${requiredCreatorVersion}\n`;
+      commandOutput += `The installed ansible-creator version on this system is ${creatorVersion}\n`;
+      commandOutput += `Please upgrade to the latest version of ansible-creator and try again.`;
+      commandResult = "failed";
+    }
+
+    await webView.postMessage({
+      command: "execution-log",
+      arguments: {
+        commandOutput: commandOutput,
+        projectUrl: destinationPathUrl,
+        status: commandResult,
+      },
+    } as PostMessageEvent);
+  }
+
+  public async runAddCommand(
+    payload: PluginFormInterface,
+    webView: vscode.Webview,
+  ) {
+    const { pluginName, pluginType, collectionPath, verbosity, isOverwritten } =
+      payload;
+    const destinationPathUrl =
+      collectionPath ||
+      `${os.homedir()}/.ansible/collections/ansible_collections`;
+
+    let ansibleCreatorAddCommand = await this.getCreatorPluginCommand(
+      pluginName,
+      pluginType.toLowerCase(),
+      destinationPathUrl,
+    );
+
+    if (isOverwritten) {
+      ansibleCreatorAddCommand += " --overwrite";
+    } else {
+      ansibleCreatorAddCommand += " --no-overwrite";
+    }
+    switch (verbosity) {
+      case "Off":
+        ansibleCreatorAddCommand += "";
+        break;
+      case "Low":
+        ansibleCreatorAddCommand += " -v";
+        break;
+      case "Medium":
+        ansibleCreatorAddCommand += " -vv";
+        break;
+      case "High":
+        ansibleCreatorAddCommand += " -vvv";
+        break;
+    }
```

Comment:
I may be misunderstanding - but is this redundant with the verbosityFlag and verbosityMap above?